sdks/go/pkg/beam/core/runtime/exec/coder.go

```diff
@@ -534,26 +1038,34 @@ func (*intervalWindowEncoder) EncodeSingle(elm typex.Window, w io.Writer) error
 
 type intervalWindowDecoder struct{}
 
-func (*intervalWindowDecoder) Decode(r io.Reader) ([]typex.Window, error) {
+func (d *intervalWindowDecoder) Decode(r io.Reader) ([]typex.Window, error) {
 	// Encoding: upper bound and duration
 
 	n, err := coder.DecodeInt32(r) // #windows
 
 	ret := make([]typex.Window, n, n)
 	for i := int32(0); i < n; i++ {
-		end, err := coder.DecodeEventTime(r)
-		if err != nil {
-			return nil, err
-		}
-		duration, err := coder.DecodeVarUint64(r)
+		w, err := d.DecodeSingle(r)
 		if err != nil {
 			return nil, err
 		}
-		ret[i] = window.IntervalWindow{Start: mtime.FromMilliseconds(end.Milliseconds() - int64(duration)), End: end}
+		ret[i] = w
 	}
 	return ret, err
 }
 
+func (*intervalWindowDecoder) DecodeSingle(r io.Reader) (typex.Window, error) {
+	end, err := coder.DecodeEventTime(r)
+	if err != nil {
+		return nil, err
+	}
+	duration, err := coder.DecodeVarInt(r)
```

Comment:
Ah no difference TBH. DecodeVarInt basically just does an extra cast from uint64 to int64, which I didn't remove. 
I went back to `DecodeVarUint64` since it avoids an extra layer of if checks and a function call.